src/components/Main/Main.tsx

```diff
@@ -0,0 +1,124 @@
+import { Todo } from '../../types/Todo';
+import cn from 'classnames';
+import { TodoItem } from '../TodoItem/TodoItem';
+import { useEffect, useRef } from 'react';
+
+interface Props {
+  todos: Todo[];
+  loader: boolean;
+  tempTodo: Todo | null;
+  deletePosts: (b: Todo) => void;
+  addActivePosts: (b: Todo) => void;
+  loadTodos: number[];
+  changeInputs: (b: Todo, c: string) => void;
+  clickTodo: number | null;
+  setChangeInput: (b: string) => void;
+  changeInput: string;
+  setClickTodo: (b: number) => void;
+}
+
+export const Main = ({
+  todos,
+  loader,
+  tempTodo,
+  deletePosts,
+  addActivePosts,
+  loadTodos,
+  changeInputs,
+  clickTodo,
+  setChangeInput,
+  changeInput,
+  setClickTodo,
+}: Props) => {
+  const inputFocus = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (clickTodo !== null && inputFocus.current) {
+      inputFocus.current.focus();
+    }
+  }, [clickTodo]);
+
+  return (
+    <section className="todoapp__main" data-cy="TodoList">
+      {todos &&
```

Comment:
Redundant check, todos is always truthy
```suggestion
      {
```